Patch-Source: https://github.com/stha09/chromium-patches/blob/cd45403f83a2305d388e1b1f930ca05c13535636/chromium-105-AdjustMaskLayerGeometry-ceilf.patch
From 385068e1eb1cde9629d18ceee1fd13255c70c806 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 22 Jul 2022 18:29:24 +0000
Subject: [PATCH] libstdc++: use math.h in blink::AdjustMaskLayerGeometry

libstdc++ does not implement std::ceilf. Use ceilf from math.h
instead.

Bug: 957519
Change-Id: I03b5e0a1eb73fdeae34d5f3d2f2e9c8871c52543
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3782841
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Reviewed-by: Juanmi Huertas <juanmihd@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1027342}
---

diff --git a/third_party/blink/renderer/platform/graphics/compositing/adjust_mask_layer_geometry.cc b/third_party/blink/renderer/platform/graphics/compositing/adjust_mask_layer_geometry.cc
index 4abe1d9..b5b43da 100644
--- a/third_party/blink/renderer/platform/graphics/compositing/adjust_mask_layer_geometry.cc
+++ b/third_party/blink/renderer/platform/graphics/compositing/adjust_mask_layer_geometry.cc
@@ -4,8 +4,9 @@
 
 #include "third_party/blink/renderer/platform/graphics/compositing/adjust_mask_layer_geometry.h"
 
+#include <math.h>
 #include <algorithm>
-#include <cmath>
+
 #include "third_party/blink/renderer/platform/graphics/paint/geometry_mapper.h"
 #include "third_party/blink/renderer/platform/graphics/paint/transform_paint_property_node.h"
 #include "ui/gfx/geometry/size.h"
@@ -29,8 +30,7 @@
   // Map a screen pixel into the layer.
   GeometryMapper::SourceToDestinationRect(TransformPaintPropertyNode::Root(),
                                           transform, pixel_rect);
-  int outset =
-      std::ceilf(std::max(pixel_rect.width(), pixel_rect.height()) * 2);
+  int outset = ceilf(std::max(pixel_rect.width(), pixel_rect.height()) * 2);
   // Don't expand too far in extreme cases.
   constexpr int kMaxOutset = 1000;
   outset = std::min(kMaxOutset, outset);
